[Feature] UI create review

Laurarestrepo03 opened the pull request:
- Finished UI for creation of reviews 
- Closes #9 
- **No** connection to DB
- **No** navigation
- Had to create a custom search bar, hence why there's no magnifying glass
- addPhotoButton could be component, however given the communication between some variables it might not be
- Categories spacing could not follow prototype, and workarounds did not work. Could make an issue out of this but it is not a priority 

https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/69609680/a4449a19-96ef-43b0-ba8e-1075894bd489

https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/69609680/89c37aec-c00b-4996-a16f-04b15343a90c

mar-cas3 reviewed (approved):
Looks amazing! 
Remember to be careful with the conflicts when merging

Laurarestrepo03 commented:
Commit 8bb7fcd5c5cbf43aec5825b01b41bd95b84116b4
**Major changes:**
- Added an x button to clear and get out of text fields (check videos at the bottom)

**Minor changes:**
- Made header padding smaller
- Deleted _next_ button from the bottom and added to the top
- Added magnifying glass to search bar

https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/69609680/680361ac-5b77-44a8-bc8b-b6825472c491


https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/69609680/1f7db80f-3a33-4a2b-9182-ab30d422d436

Note: these are screen recordings from my phone 😄

Laurarestrepo03 commented:
**COMMIT** d389e48e29ef2148c007087b38cccd4038dcf2cf

**Major changes:**
- Added camera option for review photo

https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/69609680/098ce2f8-5b05-4974-832c-c01176044554

Laurarestrepo03 commented:
@mar-cas3 @juanyepesp Hey guys, would appreciate it if you could check out my last two comments, which regard commits 8bb7fcd5c5cbf43aec5825b01b41bd95b84116b4  and d389e48e29ef2148c007087b38cccd4038dcf2cf. Let me know if there's anything I should change!

mar-cas3 commented on FoodBookApp/UI/Views/CreateReview/CreateReview1View.swift:
```diff
@@ -0,0 +1,116 @@
+    @State private var searchText: String = ""
+    @FocusState private var searchTextIsFocused: Bool
+    @State private var selectedCats: [String] = []
+    let customGray = Color(red: 242/255, green: 242/255, blue: 242/255)
```
not urgent not blocker but these colors could be in a constants or theme file later on.

mar-cas3 commented on FoodBookApp/UI/Views/CreateReview/CreateReview2View.swift:
```diff
@@ -0,0 +1,205 @@
+    @FocusState private var reviewTitleIsFocused: Bool
+    @State private var reviewBody: String = ""
+    @FocusState private var reviewBodyIsFocused: Bool
+    let customGray = Color(red: 242/255, green: 242/255, blue: 242/255)
```
Same as above

Laurarestrepo03 replied:
Definitely, and as you say it should be an issue later because I also used these colors in another branch so it's better to avoid conflicts. Thanks for the feedback!

mar-cas3 reviewed (approved):
Great job!!!
We could add an issue in the backlog (for next sprint) to separate the colors out into another file. But not a priority right noww
